jumpscale/sals/process/__init__.py

```diff
@@ -84,595 +141,833 @@ def execute(
 
 
 def is_alive(pid):
-    """Checks if pid is Running
+    """Check whether the given PID exists in the current process list.
 
-    Arguments:
-        pid (int) -- pid of the process to be checked
+    Args:
+        pid (int): Process ID (PID) to be checked.
 
     Returns:
-        [bool] -- True if process is running
+        bool: True if the given PID exists in the current process list, False otherwise.
     """
-    pid = int(pid)
     return psutil.pid_exists(pid)
 
 
 def is_installed(cmd):
-    """[summary]
-    Checks if a specific command is available on system e.g. curl
-    Arguments:
-        cmd {str} -- command to be checked
+    """Checks if a specific command is available on system e.g. curl.
+
+    Args:
+        cmd (str): Command to be checked.
 
     Returns:
-        [bool] -- True if command is installed
+        bool: True if command is available, False otherwise.
     """
-    rc, _, _ = execute("which %s" % cmd, die=False)
-    if rc:
-        return False
-    else:
-        return True
-
+    rc, _, _ = execute(f"which {cmd}", die=False)
+    return rc == 0
 
-def kill(pid, sig=signal.SIGTERM.value):
-    """Kill a process with a signal
 
-    Arguments:
-        pid (int) -- pid of the process to be killed
+def kill(proc, sig=signal.SIGTERM, timeout=5, sure_kill=False):
```

Comment:
`sure_kill` can be renamed to `force` or something like `graceful`?

Reply:
now you ruined my day :D
I like the `force_kill` name alot for that option for two reasons.

**first**, the name is so obvious, you want to be sure that the process will be killed.
**second** and the most important, in the *nix universe, the `-9` option for the `kill` command, or `SIGKILL` option often referred to as the “sure kill” option, this where I came up with this name in the first place.

Reply:
yes, that's a suggestion, you can use any proper name for sure :D